core/variant_call.cpp

```diff
@@ -1204,7 +1204,7 @@ Variant Variant::construct(const Variant::Type p_type, const Variant **p_args, i
 				return (Transform(p_args[0]->operator Transform()));
 
 			// misc types
-			case COLOR: return p_args[0]->type == Variant::STRING ? Color::html(*p_args[0]) : Color::hex(*p_args[0]);
+			case COLOR: return p_args[0]->type == Variant::STRING ? Color::html(*p_args[0]) : Color();
```

Comment:
I dont think this should be removed. It is not a random result. It was a very intentional choice to allow users to construct a color from a single int. See the documentation here http://docs.godotengine.org/en/latest/classes/class_color.html#class-color-method-color

Reply:
I have updated the logic so that it will use the old one when converting from INT => Color.